### PR TITLE
add timeout to HTTP requests

### PR DESCRIPTION
**Pull request by baris**

### CODE REVIEW

File: medium.go [hunk 1]
```diff
@@ -143,6 +146,7 @@ type Medium struct {
 	ApplicationSecret string
 	AccessToken       string
 	Host              string
+	Timeout           int
```
Comment (majelbstoat) [inline medium.go]: Can you make this type time.Duration. (defaultTimeout `5 * time.Second`)

**Comment by majelbstoat** [comment]
Thanks for this submission, I'm inclined to accept. @baris, can you fill out a CLA per the readme? cc: @dpup 2nd eyes.

File: medium.go (hunk 1 above)
Comment (baris) [inline medium.go]: Yes, sure. Being explicit on the type makes the API more flexible. Updating the PR shortly.

**Comment by baris** [comment]
This branch is now fixing #8 too (b5831d4). It was a single line change in the regex and was annoying enough that I went ahead and included in this branch.

**Comment by andybons** [comment]
Great! Thank you for doing that.

Handing off to @majelbstoat and @dpup.

**Comment by majelbstoat** [comment]
@baris, this looks good. Can you squash to a single commit (or two, if you want to preserve the flaky test fix as a separate one), and sign the CLA in github.com/medium/opensource, then we can merge.

**Comment by baris** [comment]
@majelbstoat, I squashed the commits into a single one. I'll check and sign the CLA as soon as possible.

**Comment by majelbstoat** [comment]
I get the following errors when running `go test` with this patch applied:

```
--- FAIL: TestAPITimeout (0.00s)
	medium_test.go:97: Expected HTTP timeout error, got medium: Failed to make request: Get http://127.0.0.1:59094/v1/me: read tcp 127.0.0.1:59094: use of closed network connection (-1)
FAIL
exit status 1
FAIL	_/Users/jamie/Development/workspace/medium-sdk-go	0.011s
```

**Comment by baris** [comment]
I'm using `go version devel +50d0ee0 Thu Sep 17 01:20:45 2015 +0000 linux/amd64` and it works for me.

````
-->  (http_timeout) go test
PASS
ok      _/home/baris/repos/medium-sdk-go        0.009s
```

but I see that the test is failing on 1.4.

```
-->  (http_timeout) PATH=/home/baris/src/go-1.4/bin:$PATH GOROOT=/home/baris/src/go-1.4/ go test
--- FAIL: TestAPITimeout (0.00s)
        medium_test.go:97: Expected HTTP timeout error, got medium: Failed to make request: Get http://127.0.0.1:51500/v1/me: read tcp 127.0.0.1:51500: use of closed network connection (-1)
FAIL
exit status 1
FAIL    _/home/baris/repos/medium-sdk-go        0.015s
```

Let me take a look.

**Comment by baris** [comment]
Apparently go1.4 doesn't set the timeout error. I've updated the test accordingly.

**Comment by majelbstoat** [comment]
Cool. Tests pass locally now. Merging once CLA is confirmed.

**Comment by majelbstoat** [comment]
Hey @baris, looking forward to merging this. Just wanted to confirm you'd signed the CLA at github.com/medium/opensource, before I do?

**Comment by baris** [comment]
Hi @majelbstoat. I'd forgot about this one. Sorry. I just signed it.

**Pull request merged**